### PR TITLE
Feat/devsu 2275 make summary type configurable

### PR DESCRIPTION
**Pull request by elewis2**
Makes summary type selectable when creating a new report. This allows users to create new reports using eg the rapid report summary instead of the genomic report summary.

### CODE REVIEW

File: app/views/ReportView/components/Summary/index.tsx [hunk 1]
```diff
@@ -11,36 +11,43 @@ const Summary = ({
   templateName,
   ...props
 }: SummaryProps): JSX.Element => {
-  if (templateName === 'probe') {
+  // TODO remove backup template name checks when data is updateed in prod
+  const summarySection = props.visibleSections.find(element => ['summary-genomic', 'summary-tgr', 'summary-pcp', 'summary-probe'].includes(element));
```
Comment (bnguyen-bcgsc) [inline app/views/ReportView/components/Summary/index.tsx:15]: I got an error with this line when trying to print the report: "Cannot read properties of undefined (reading 'find').
TypeError: Cannot read properties of undefined (reading 'find')"

File: app/views/ReportView/components/Summary/index.tsx [hunk 2]
```diff
@@ -11,36 +11,43 @@ const Summary = ({
...
+  if (summarySection === 'summary-probe' || templateName === 'probe') {
     return (
       <ProbeSummary {...props} />
     );
   }
 
-  if (templateName === 'pharmacogenomic') {
+  if (summarySection === 'summary-pcp' || templateName === 'pharmacogenomic') {
+    console.log(templateName);
```
Comment (bnguyen-bcgsc) [inline app/views/ReportView/components/Summary/index.tsx:23]: Are we keeping the console logs on purpose?

**Review by bnguyen-bcgsc — CHANGES_REQUESTED** [review]

File: app/views/ReportView/components/Summary/index.tsx (hunk 2 above)
Comment (elewis2) [reply app/views/ReportView/components/Summary/index.tsx:23]: ah, no

File: app/views/ReportView/components/Summary/index.tsx (hunk 1 above)
Comment (elewis2) [reply app/views/ReportView/components/Summary/index.tsx]: fixed, had to update the SummaryProps type

**Review by bnguyen-bcgsc — APPROVED** [review]

**Pull request merged**